pkg/controller/errors.go

```diff
@@ -12,15 +14,27 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const optimisticLockErrorMsg = "the object has been modified; please apply your changes to the latest version and try again"
+
 type causer interface {
 	Cause() error
 }
 
+func IsOptimisticLockError(err error) bool {
```

Comment:
Can this error be checked by errors.IsConflict(error) in the api/errors package?